addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/stick/TransmitterStick.java

```diff
@@ -0,0 +1,402 @@
+                    if (existingCmd.getPriority() < newCmd.priority) {
+                        // we have an older INFO command with same or lower priority, remove
+                        queuedCommands.remove();
+                    } else {
+                        // existing has higher priority, skip addition
+                        return false;
+                    }
+                }
+
+                if (newCmd.getCommandType() != CommandType.INFO && existingCmd.getCommandType() != CommandType.INFO) {
+                    // we have an older command for the same channels, remove
+                    queuedCommands.remove();
+                }
+            }
+        }
+
+        return true;
+    }
+
+    static class DueCommandSet extends TreeSet<Command> {
+        private static final long serialVersionUID = -3216360253151368826L;
+
+        public DueCommandSet() {
+            super(new Comparator<Command>() {
+                /**
+                 * Due commands are sorted by priority first and then by delay.
+                 */
+                @Override
+                public int compare(Command o1, Command o2) {
+                    if (o1.equals(o2)) {
+                        return 0;
+                    }
+
+                    int d = o2.getPriority() - o1.getPriority();
+                    if (d < 0) {
+                        return -1;
+                    }
+
+                    if (d == 0 && o1.getDelay(TimeUnit.MILLISECONDS) < o2.getDelay(TimeUnit.MILLISECONDS)) {
+                        return -1;
+                    }
+                    return 1;
+                }
+            });
+        }
+
+        @Override
+        public boolean add(Command e) {
+            if (TransmitterStick.prepareAddition(e, this)) {
+                return super.add(e);
+            }
+
+            return false;
+        }
+    }
+
+    class CommandWorker implements Runnable {
+        private ArrayList<Integer> validIds = new ArrayList<>();
+        private final AtomicBoolean terminated = new AtomicBoolean();
+        private final int updateInterval;
+        private final SerialConnection connection;
+
+        private final BlockingQueue<Command> cmdQueue = new DelayQueue<Command>() {
+            @Override
+            public boolean add(Command e) {
+                if (TransmitterStick.prepareAddition(e, this)) {
+                    return super.add(e);
+                }
+
+                return false;
+            }
+        };
+
+        CommandWorker() {
+            connection = new SerialConnection(config.portName);
+            updateInterval = config.updateInterval;
+        }
+
+        void terminateUpdates() {
+            terminated.set(true);
+
+            // add a NONE command to make the thread exit from the call to take()
+            cmdQueue.add(new Command(CommandType.NONE));
+        }
+
+        void requestUpdate(Integer... channelIds) {
```

Comment:
Wouldn't it be better to use `void requestUpdate(List<Integer> channelIds)`? because then the callers can pass their ArrayLists directly.